### PR TITLE
Stub varlink pod methods.

### PR DESCRIPTION
**Pull request by haircommander**
Signed-off-by: haircommander <pehunt@redhat.com>

### CODE REVIEW

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 1]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# #### Example
+# ~~~
+# $ varlink call unix:/run/podman/io.projectatomic.podman/io.projectatomic.podman.CreatePod '{"name": "test"}'
+# {
+#   "pod": "8759dafbc0a4dc3bcfb57eeb72e4331eb73c5cc09ab968e65ce45b9ad5c4b6bb"
+# }
+# ~~~
+method CreatePod() -> (notimplemented: NotImplemented)
+
+# ListPods returns a list of pods in no particular order.  There are
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: They are ...

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 2]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# returned as an array of ListPodData structs.  See also [GetPod](#GetPod).
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: link to ListPodData

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 3]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+method ListPod() -> (notimplemented: NotImplemented)
+
+# GetPod takes a name or ID of a pod and returns single ListPodData
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: link to the struct in the API doc.

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 4]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# structure.  A [PodNotFound](#PodNotFound) error will be returned if the pod cannot be found.
+# See also [ListPods](ListPods).
+method GetPod() -> (notimplemented: NotImplemented)
+
+# InspectPod takes the ID or name of a pod and returns one InspectPodData
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: Link struct

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 5]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# ~~~
+# $ varlink call -m unix:/run/podman/io.projectatomic.podman/io.projectatomic.podman.StopPod '{"name": "135d71b9495f"}'
+# {
+#   "pod": "135d71b9495f7c3967f536edad57750bfdb569336cd107d8aabab45565ffcfb6"
+# }
+# ~~~
+method StopPod() -> (notimplemented: NotImplemented)
+
+# RestartPod will restart containers in a pod given a pod name or ID. Containers in 
+# the pod that are running will be stopped, then all stopped containers will be run.
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: run-on

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 6]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# ~~~
+# $ varlink call -m unix:/run/podman/io.projectatomic.podman/io.projectatomic.podman.RestartPod '{"name": "135d71b9495f"}'
+# {
+#   "pod": "135d71b9495f7c3967f536edad57750bfdb569336cd107d8aabab45565ffcfb6"
+# }
+# ~~~
+method RestartPod() -> (notimplemented: NotImplemented)
+
+# KillPod takes the name or ID of a pod as well as a signal to be applied to the pod.  Once the
+# containers in the pod have been killed, the pod's ID is returned.  If the pod cannot be found, a
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: long term, it is better to not use the apostrophe possessive ... "the ID of the pod"

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 7]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# See also [UnpausePod](#UnpausePod).
+method PausePod() -> (notimplemented: NotImplemented)
+
+# UnpausePod takes the name or ID of a pod and unpauses the paused containers associated with it.  If the pod cannot be
+# found, a [PodNotFound](#PodNotFound) error will be returned; otherwise the ID of the pod is returned.
+# See also [PausePod](#PausePod).
+method UnpausePod() -> (notimplemented: NotImplemented)
+
+# RemovePod takes the name or ID of a pod as well a boolean representing whether a running
+# container in the pod can be stopped and removed.  If a pod has containers associated with it, and force is not true
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: add trailing comma

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 8]
```diff
@@ -609,13 +629,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
 # ImageNotFound means the image could not be found by the provided name or ID in local storage.
 error ImageNotFound (name: string)
 
 # ContainerNotFound means the container could not be found by the provided name or ID in local storage.
 error ContainerNotFound (name: string)
 
+# PodNotFound means the pod could not be found by the provided name or ID in local storage.
+error PodNotFound (name: string)
+
+# PodNoContainers means the pod does not have containers associated and cannot be started or restarted.
```
Comment (baude) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: does not have any containers and cannot ... ?

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 9]
```diff
@@ -609,13 +642,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# $ varlink call unix:/run/podman/io.projectatomic.podman/io.projectatomic.podman.CreatePod '{"name": "test"}'
+# {
+#   "pod": "8759dafbc0a4dc3bcfb57eeb72e4331eb73c5cc09ab968e65ce45b9ad5c4b6bb"
+# }
+# ~~~
+method CreatePod() -> (notimplemented: NotImplemented)
+
+# ListPods returns a list of pods in no particular order.  They are
+# returned as an array of ListPodData structs.  See also [GetPod](#GetPod).
+method ListPod() -> (notimplemented: NotImplemented)
```
Comment (jwhonce) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: Is method ListPods() or ListPod()?  I vote for ListPods().

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 10]
```diff
@@ -609,13 +642,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+# GetPod takes a name or ID of a pod and returns single [ListPodData](#ListPodData)
+# structure.  A [PodNotFound](#PodNotFound) error will be returned if the pod cannot be found.
+# See also [ListPods](ListPods).
+method GetPod() -> (notimplemented: NotImplemented)
+
+# InspectPod takes the ID or name of a pod and returns one [InspectPodData](#InspectPodData)
+# structure.  A [PodNotFound](#PodNotFound) error will be returned if the pod cannot be found.
+# See also [ListPods](ListPods).
+# Note: this structure is not implemented because podman pod inspect has not been implemented.
+method InspectPod() -> (notimplemented: NotImplemented)
```
Comment (jwhonce) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: InspectImage() and InspectContainer() return JSON blobs.  InspectPod is going to be a struct?

File: cmd/podman/varlink/io.projectatomic.podman.varlink [hunk 11]
```diff
@@ -609,13 +642,126 @@ method ExportImage(name: string, destination: string, compress: bool, tags: []st
...
+#   "pod": "135d71b9495f7c3967f536edad57750bfdb569336cd107d8aabab45565ffcfb6"
+# }
+# ~~~
+method StopPod() -> (notimplemented: NotImplemented)
+
+# RestartPod will restart containers in a pod given a pod name or ID. Containers in
+# the pod that are running will be stopped, then all stopped containers will be run.
+# If the pod cannot be found by name or ID, a [PodNotFound](#PodNotFound) error will be
+# returned.  If the pod has no containers, a [PodNoContainers](#PodNoContainers)
+# error will be returned.  If the pod is successfully started, the pod ID will be returned.
```
Comment (jwhonce) [inline cmd/podman/varlink/io.projectatomic.podman.varlink]: What is returned if you have a partial start?

**Comment by baude** [comment]
bot, retest this please

**Comment by mheon** [comment]
@baude @jwhonce PTAL

**Comment by jwhonce** [comment]
LGTM

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**